pg_backup_api/pg_backup_api/utils.py

```diff
@@ -76,3 +77,14 @@ def get_server_by_name(server_name):
         conf = barman.__config__.get_server(server)
         if server == server_name:
             return conf
+
+
+def parse_backup_id(server, backup_id):
```

Comment:
Is there any reason why we can't use `barman.cli.parse_backup_id`? If we can't use it as it is then I think it's worth patching barman to address whatever prevents its reuse so we can avoid duplicating the code here in future versions.

Reply:
barman.output doesn't get along with Flask making it hangs when `output.error_exit_code` is called.

We need to move the barman.cli.parse_backup_id's logic out into utils, that way we could import from there

Reply:
Makes sense - we'll update that in the next Barman release then we can remove the duplication in a subsequent pg-backup-api version.